src/common/middleware/call-api.js

```diff
@@ -50,39 +50,26 @@ export default defaults => () => next => action => {
                 }));
               }
 
-              return Promise.resolve(result);
+              // resolve promise with response JSON
+              return result;
             });
-        })
-        .catch((error) => {
```

Comment:
Also removed duplicated error handling.

We already have a `catch` on top level fetch promise, so it will catch the errors from nested promises.